dashboard/app/models/levels/ailab.rb

```diff
@@ -10,12 +10,12 @@
 #  level_num             :string(255)
 #  ideal_level_source_id :bigint           unsigned
 #  user_id               :integer
-#  properties            :text(16777215)
+#  properties            :text(4294967295)
 #  type                  :string(255)
 #  md5                   :string(255)
 #  published             :boolean          default(FALSE), not null
-#  notes                 :text(65535)
-#  audit_log             :text(65535)
+#  notes                 :text(16777215)
+#  audit_log             :text(16777215)
```

Comment:
These were generated by the migration, although I'm not sure why it's so much bigger (The documentation implies 4x bigger: 65,535*4=262,140 but these new values are 256 times bigger?!). See documentation here: https://dev.mysql.com/doc/refman/8.0/en/alter-table.html

Reply:
just curious, where do you see 4x bigger? naively, I would expect the size needs to increase by 33%, from 3 bytes / char to 4 bytes / char in the worst case.

in any case, yes, this is a much bigger jump! it looks like mysql is just going up to the next string size which it calls `mediumtext`. so I think this is expected.

Reply:
Ah yes, this was the paragraph I was using from the doc:

"For a column that has a data type of VARCHAR or one of the TEXT types, CONVERT TO CHARACTER SET changes the data type as necessary to ensure that the new column is long enough to store as many characters as the original column. For example, a TEXT column has two length bytes, which store the byte-length of values in the column, up to a maximum of 65,535. For a latin1 TEXT column, each character requires a single byte, so the column can store up to 65,535 characters. If the column is converted to utf8, each character might require up to three bytes, for a maximum possible length of 3 × 65,535 = 196,605 bytes. That length does not fit in a TEXT column's length bytes, so MySQL converts the data type to MEDIUMTEXT, which is the smallest string type for which the length bytes can record a value of 196,605. Similarly, a VARCHAR column might be converted to MEDIUMTEXT."

But that's going from 1 -> 3 bytes/character, but we're doing 3 -> 4 bytes/character. What you're saying makes sense!